syntax = "proto3";

service ExecAction {

   rpc RunAction(ActionList) returns (stream ActionStatus) {}

}

message ActionList {
  int32 count=1;
  repeated Action actions=2;
}

message Action {
    string title=1;
    string code=2;
}

message ActionStatus {
    int32 id=1;
    int32 current_step=2;
    int32 total_steps=3;
}
